tests/test_types_typeddict.py

```diff
@@ -378,3 +378,111 @@ class Model(BaseModel):
             Model(d=input_value)
     else:
         assert Model(d=input_value).d == expected
+
+
+class RecursiveTypedDict(typing_extensions.TypedDict):
+    # TODO: See if we can get this working if defined in a function (right now, needs to be module-level)
```

Comment:
you need to use `create_module` as is done lots in `test_forwards_refs.py`.

Reply:
That works, but I also think it _may_ be possible to get it working inside a function and not a module. I asked Adrian to add this comment so I could look at it later when I look at the other recursive/generic stuff. Module is a good idea though to prevent it from running on test execution (making it easier to prevent debugger from hitting unwanted code)